### PR TITLE
Add nostr wallet auth

### PR DESCRIPTION
**Pull request by futurepaul**

### CODE REVIEW

File: src/api/walletAuth.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        return data;
+    } catch (e) {
+        console.error(e);
+        throw new Error(eify(e).message);
+    }
+}
+
+// async function handleConnectNWA() {
+//   console.log("handleConnectNWA")
```
Comment (benthecarman) [inline src/api/walletAuth.ts]: Can't remove all this

File: src/api/walletAuth.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
+import { eify } from "~/utils/eify";
+
+const API_URL = import.meta.env.VITE_ZAPPLE_API_URL;
+
+export async function fetchNwaStatus(
+    nwaResponse: { id: string; uri: string },
+    _lastResponse: any
```
Comment (benthecarman) [inline src/api/walletAuth.ts]: Have this unused param in a few functions

File: src/components/IntegratedQr.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            <span class="truncate">{props.text}</span>
+            <span class="pr-2">
+                {props.text.length > 32 ? props.text.slice(-8) : ""}
+            </span>
+        </div>
+    );
+}
+
+async function share(receiveString: string) {
+    // If the browser doesn't support share we can just copy the address
```
Comment (benthecarman) [inline src/components/IntegratedQr.tsx]: You're not copying here, think you can remove this comment

File: src/components/IntegratedQr.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    copied: Accessor<boolean>
+] => {
+    const [copied, setCopied] = createSignal(false);
+    let timeout: ReturnType<typeof setTimeout> | undefined;
+    const copy: CopyFn = async (text) => {
+
+            await navigator.clipboard.writeText(text);
+        setCopied(true);
+        if (timeout) clearTimeout(timeout);
+        timeout = setTimeout(() => setCopied(false), copiedTimeout);
```
Comment (benthecarman) [inline src/components/IntegratedQr.tsx]: Could just be my phone but the formatting looks weird here

**Comment by futurepaul** [comment]
@benthecarman addressed your comments ty

File: src/api/walletAuth.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        const data = await response.json();
+
+        return data;
+    } catch (e) {
+        console.error(e);
+        throw new Error(eify(e).message);
+    }
+}
+
+// async function handleConnectNWA() {
```
Comment (benthecarman) [inline src/api/walletAuth.ts]: this commented out code is still here

Comment (futurepaul) [reply src/api/walletAuth.ts]: sry, pushed now

**Review by benthecarman — APPROVED** [review]

**Comment by benthecarman** [comment]
Seems like it never sends the emoji configured so the backend is always defaulting to :zap:

**Pull request merged**